.cursor/rules/go-ec-api-architecture.mdc

```diff
@@ -0,0 +1,111 @@
+---
+description: 
+globs: api/*.go
+alwaysApply: false
+---
+# Go Embedded Cluster API Implementation Guidelines
+
+For comprehensive architecture and package structure details, see [api/README.md](mdc:../api/README.md).
+
+## Essential Implementation Patterns
+
+### Error Handling
+
+#### Structured API Errors
+
+Use structured error types for APIs that need to return detailed error information:
+
+```go
+type APIError struct {
+    StatusCode int         `json:"status_code,omitempty"`
+    Message    string      `json:"message"`
+    Field      string      `json:"field,omitempty"`
+    Errors     []*APIError `json:"errors,omitempty"`
+    err        error       `json:"-"`
+}
+```
+
+#### Error Constructor Functions
+
+Create constructor functions for common API error types.
+
+#### API Error Handling Patterns
+
+- Always log errors with context: `a.logError(r, err, "descriptive message")`
+- Use structured errors: `types.NewBadRequestError(err)`, `types.NewInternalServerError(err)`
+- Wrap errors with context: `fmt.Errorf("operation failed: %w", err)`
+
+#### Specific API Error Types
+
+- **Use specific API error types**: Always use `NewBadRequestError`, `NewInternalServerError`, etc. instead of generic errors for proper HTTP status codes
+  ```go
+  return NewBadRequestError(errors.New("invalid input"))  // Good
+  return errors.New("invalid input")                      // Bad
+  ```
+
+- **Return consistent HTTP status codes**: Use appropriate APIError types for different error conditions
+  ```go
+  return NewBadRequestError(errors.New("invalid input"))        // 400
+  return NewForbiddenError(errors.New("access denied"))         // 403
+  return NewInternalServerError(errors.New("database failed"))  // 500
+  ```
+
+### HTTP Handler Patterns
+
+- Use method receivers on the main API struct: `func (a *API) handlerName(...)`
```

Comment:
The handler no longer hang off the API struct. There are dedicated Handler structs now.

The names include the HTTP method, e.g. `func (h *Handler) PostConfigureInstallation(w http.ResponseWriter, r *http.Request) {`

```suggestion
- Use method receivers on the main API struct: `func (a *API) handlerName(...)`
```